scheduler/job/job.go

```diff
@@ -297,3 +314,163 @@ func (j *job) syncPeers() (string, error) {
+		return "", err
+	}
+
+	// Get all peers by task id
+	peers, err := j.getValidPeers(req.TaskID)
+	if err != nil {
+		logger.Errorf("get peers by task id %s failed: %s", req.TaskID, err.Error())
+		return "", err
+	}
+
+	// Delete task by task id and host id
+	successTasks := make([]*internaljob.Task, 0)
+	failureTasks := make([]*internaljob.Task, 0)
+
+	// Create a wait group to limit delete rpc concurrency
+	// and avoid too many rpc requests to the host.
+	wg := sync.WaitGroup{}
+	deleteTaskLimit := make(chan struct{}, deleteTaskConcurrency)
+	for _, peer := range peers {
+		wg.Add(1)
+		deleteTaskLimit <- struct{}{}
+		go func(peer *resource.Peer) {
+			defer func() {
+				wg.Done()
+				<-deleteTaskLimit
+			}()
+
+			// Get dfdaemon client from host
+			target := fmt.Sprintf("%s:%d", peer.Host.IP, peer.Host.Port)
+			conn, err := grpc.DialContext(
+				ctx,
+				target,
+				grpc.WithIdleTimeout(0),
+				grpc.WithDefaultCallOptions(
+					grpc.MaxCallRecvMsgSize(math.MaxInt32),
+					grpc.MaxCallSendMsgSize(math.MaxInt32),
+				),
+				grpc.WithUnaryInterceptor(grpc_middleware.ChainUnaryClient(
+					grpc_prometheus.UnaryClientInterceptor,
+					grpc_zap.UnaryClientInterceptor(logger.GrpcLogger.Desugar()),
+					grpc_retry.UnaryClientInterceptor(
+						grpc_retry.WithMax(deleteTaskMaxRetries),
+						grpc_retry.WithBackoff(grpc_retry.BackoffLinear(deleteTaskBackoffWaitBetween)),
+					),
+				)),
+			)
+			if err != nil {
+				logger.Errorf("create grpc client to %s failed: %s", target, err.Error())
+				failureTasks = append(failureTasks, &internaljob.Task{
+					Task:        peer.Task,
+					Peer:        peer,
+					Description: err.Error(),
+				})
+				return
+			}
+
+			dfdaemonUploadClient := dfdaemonv2.NewDfdaemonUploadClient(conn)
+			_, err = dfdaemonUploadClient.DeleteCacheTask(ctx, &dfdaemonv2.DeleteCacheTaskRequest{
+				TaskId: req.TaskID,
+			})
+			if err != nil {
+				logger.Errorf("delete task %s from %s failed: %s", req.TaskID, target, err.Error())
+				failureTasks = append(failureTasks, &internaljob.Task{
+					Task:        peer.Task,
+					Peer:        peer,
+					Description: err.Error(),
+				})
+				return
+			}
+
+			successTasks = append(successTasks, &internaljob.Task{
+				Task:        peer.Task,
+				Peer:        peer,
+				Description: fmt.Sprintf("delete task %s from %s success", req.TaskID, target),
+			})
+		}(peer)
+	}
+
+	wg.Wait()
+
+	deleteTaskResponse := &internaljob.DeleteTaskResponse{
+		SuccessTasks: successTasks,
+		FailureTasks: failureTasks,
+	}
+
+	return internaljob.MarshalResponse(deleteTaskResponse)
+}
+
+// getValidPeers try to get valid peers by task id
+func (j *job) getValidPeers(taskID string) ([]*resource.Peer, error) {
+	// get task info by task id
+	task, ok := j.resource.TaskManager().Load(taskID)
+	if !ok {
+		logger.Errorf("task %s not found", taskID)
+		return nil, fmt.Errorf("task %s not found", taskID)
+	}
+
+	// get peer info by task info
+	peers := make([]*resource.Peer, 0)
```

Comment:
Add `LoadFinishedPeers` to task.

Reply:
OK, moved to `scheduler/resource/task.go::LoadFinishedPeers`.